homeassistant/components/metoffice/sensor.py

```diff
@@ -30,175 +32,163 @@
 }
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_API_KEY): cv.string,
-        vol.Required(CONF_MONITORED_CONDITIONS, default=[]): vol.All(
-            cv.ensure_list, [vol.In(SENSOR_TYPES)]
-        ),
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Inclusive(
-            CONF_LATITUDE, "coordinates", "Latitude and longitude must exist together"
-        ): cv.latitude,
-        vol.Inclusive(
-            CONF_LONGITUDE, "coordinates", "Latitude and longitude must exist together"
-        ): cv.longitude,
-    }
-)
-
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Met Office sensor platform."""
-    api_key = config.get(CONF_API_KEY)
-    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
-    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
-    name = config.get(CONF_NAME)
-
-    datapoint = dp.connection(api_key=api_key)
-
-    if None in (latitude, longitude):
-        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
-        return
 
-    try:
-        site = datapoint.get_nearest_site(latitude=latitude, longitude=longitude)
-    except dp.exceptions.APIException as err:
-        _LOGGER.error("Received error from Met Office Datapoint: %s", err)
-        return
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigType, async_add_entities
+) -> None:
+    """Set up the Met Office weather sensor platform."""
+    hass_data = hass.data[DOMAIN][entry.entry_id]
 
-    if not site:
-        _LOGGER.error("Unable to get nearest Met Office forecast site")
-        return
-
-    data = MetOfficeCurrentData(hass, datapoint, site)
-    data.update()
-    if data.data is None:
-        return
-
-    sensors = []
-    for variable in config[CONF_MONITORED_CONDITIONS]:
-        sensors.append(MetOfficeCurrentSensor(site, data, variable, name))
-
-    add_entities(sensors, True)
+    async_add_entities(
+        [
+            MetOfficeCurrentSensor(entry.data, hass_data, sensor_type)
+            for sensor_type in SENSOR_TYPES
+        ],
+        False,
+    )
 
 
 class MetOfficeCurrentSensor(Entity):
-    """Implementation of a Met Office current sensor."""
+    """Implementation of a Met Office current weather condition sensor."""
 
-    def __init__(self, site, data, condition, name):
+    def __init__(self, entry_data, hass_data, sensor_type):
         """Initialize the sensor."""
-        self._condition = condition
-        self.data = data
-        self._name = name
-        self.site = site
+        self._data = hass_data[METOFFICE_DATA]
+        self._coordinator = hass_data[METOFFICE_COORDINATOR]
+
+        self._type = sensor_type
+        self._name = f"{hass_data[METOFFICE_NAME]} {SENSOR_TYPES[self._type][0]}"
+        self._unique_id = f"{hass_data[METOFFICE_NAME]}_{SENSOR_TYPES[self._type][0]}_{self._data.latitude}_{self._data.longitude}"
```

Comment:
The configured name should not be part of the unique id. Latitude + longitude + sensor type is enough.

Reply:
No problem.